model_archiver/model_packaging_utils.py

```diff
@@ -0,0 +1,288 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""
+Helper utils for Model Export tool
+"""
+
+import os
+import json
+import zipfile
+import sys
+import logging
+import re
+from .manifest_components.publisher import Publisher
+from .manifest_components.manifest import Manifest
+from .manifest_components.engine import Engine
+from .manifest_components.model import Model
+
+MODEL_ARCHIVE_EXTENSION = '.mar'
+MODEL_SERVER_VERSION = '1.0'
+MODEL_ARCHIVE_VERSION = '1.0'
+MANIFEST_FILE_NAME = 'MANIFEST.json'
+MAR_INF = 'MAR-INF'
+ONNX_TYPE = '.onnx'
+
+
+class ModelExportUtils(object):
+    """
+    Helper utils for Model Export tool.
+    This class lists out all the methods such as validations for model archiving, ONNX model checking etc.
+    This is to keep the code in export_model.py clean and simple.
+    """
+
+    @staticmethod
+    def check_mar_already_exists(model_name, export_file_path, overwrite):
+        """
+        Function to check if .mar already exists
+        :param model_name:
+        :param export_file_path:
+        :param overwrite:
+        :return:
+        """
+        if export_file_path is None:
+            export_file_path = os.getcwd()
+
+        export_file = os.path.join(export_file_path, '{}{}'.format(model_name, MODEL_ARCHIVE_EXTENSION))
+
+        if os.path.exists(export_file):
+            if overwrite:
+                logging.warning("%s already exists. It will be overwritten since --force/-f was specified", export_file)
+
+            else:
+                logging.error("%s already exists. Since no --force/-f was specified, it will not be overwritten. "
+                              "Exiting the program here. Specify --force/-f flag to overwrite the %s file. "
+                              "See -h/--help for more details", export_file, export_file)
+
+                sys.exit(1)
```

Comment:
It's OK to sue sys.exit() in this project. But in general, using sys.exit() isn't a good practice.

Reply:
Got it.